woocommerce/woocommerce-ios
# Edit Product: update inventory settings remotely

Remote update for #1424 

## Summary

This PR enabled updating the Product inventory settings remotely.

## Changes

- When encoding a `Product` in the Networking layer, included the 6 inventory properties
- Created `ProductInventoryEditableData` struct that contains the 6 inventory properties so that each update function isn't crazy long
- Passed a completion block to `ProductInventorySettingsViewController` when the user taps "Done" on the Inventory Settings screen
- Added `inventorySettingsUpdated` to `ProductUpdater` protocol, and implemented it in the `Product` extension
- Updated Product's inventory settings locally in `ProductFormViewController` when the user completes updating the inventory settings
- Updated unit tests

## Testing

- Launch the app
- Go to Products tab
- Tap on a simple Product
- Tap "Edit" in the navigation bar
- Tap the Inventory settings row
- Change the SKU
- Toggle the "Manage stock", and update the fields
- Toggle the "Limit one per order" switch
- Tap "Done" --> the fields in the Inventory settings row should be updated
- Tap "Done" on the Edit Product main screen --> the Product inventory settings should be updated on wp-admin
- If possible, repeat the above steps with a different "Manage stock" value and confirm the properties are updated on wp-admin

Update release notes:
